.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,23 @@
+
+name: 'Shellcheck'
+
+on:
+  push:
+    paths:
+      - '**.sh'
+
+  pull_request:
+    paths:
+      - '**.sh'
+
+permissions:
+  contents: read
+
+jobs:
+  check-scripts:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Is it possible to recurse through every directory and do the checks rather than manually specifying the paths?

Reply:
I'm not sure how to do this. Can investigate further

Reply:
So run: by default uses bash. It may be possible to use the `globstar` pattern to recurse:

i.e. `shopt -s globstar && shellcheck **/*.sh`